tests/utils_tests.py

```diff
@@ -1162,3 +1164,15 @@ def test_get_form_data_token(self):
         assert get_form_data_token({"token": "token_abcdefg1"}) == "token_abcdefg1"
         generated_token = get_form_data_token({})
         assert re.match(r"^token_[a-z0-9]{8}$", generated_token) is not None
+
+    def test_get_database_ids(self) -> None:
+        world_health = db.session.query(Dashboard).filter_by(slug="world_health").one()
```

Comment:
let's tests some edge cases here:
1. there is no database behind it e.g. markdown only
2. there are not slices in the dashboard
3. there are 2 databases used for the dashboard

Reply:
1. there is no `markdown slice` any more, 1 and 2 are same case: no slices
Added test cases for no slice, 2 database and druid slice cases.

Reply:
wow nice, what about iframe chart?

Reply:
`markup`, `iframe` and `separator` visualization types, are not real slices since they do not generate query, i converted them into dashboard component and retried these viz types.
https://github.com/apache/incubator-superset/pull/10590
https://github.com/apache-superset/superset-ui/pull/746